track.go

```diff
@@ -108,6 +108,7 @@ func newVideoTrack(opts *MediaDevicesOptions, d driver.Driver, constraints Media
 	vr := d.(driver.VideoRecorder)
 	r, err := vr.VideoRecord(constraints.Media)
 	if err != nil {
+		_ = d.Close()
```

Comment:
I think we should just Close the driver automatically in `driver` package. And, we should simply call the wrapper version of Close so the driver state will be updated too and finally still return the error. It looks to me that this is the `driver` package problem. What do you think?

Reply:
Ah, yes. This can be done in the wrapper.

Reply:
![image](https://user-images.githubusercontent.com/8390204/74899786-3464f500-53e1-11ea-95c9-1ba37ed8489f.png)
So, there will be a choice pseudo-state at the place of `Audio/VideoRecord` and an another one arrow to `Closed` state on failure case.

Reply:
Makes sense. Thanks for pointing that out!